pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/merge_rollup/MergeRollupTaskExecutor.java

```diff
@@ -64,27 +57,29 @@
     LOGGER.info("Starting task: {} with configs: {}", taskType, configs);
     long startMillis = System.currentTimeMillis();
 
-    Preconditions.checkState(
-        MergeType.CONCAT.name().equalsIgnoreCase(configs.get(MinionConstants.MergeRollupTask.MERGE_TYPE_KEY)),
-        "Only 'CONCAT' mode is currently supported.");
-
     String tableNameWithType = configs.get(MinionConstants.TABLE_NAME_KEY);
     TableConfig tableConfig = getTableConfig(tableNameWithType);
     Schema schema = getSchema(tableNameWithType);
 
-    Map<String, AggregationFunctionType> aggregationTypes = MergeRollupTaskUtils.getRollupAggregationTypes(configs);
-    String numRecordsPerSegmentString = configs.get(MinionConstants.MergeRollupTask.MAX_NUM_RECORDS_PER_SEGMENT);
-
     SegmentProcessorConfig.Builder segmentProcessorConfigBuilder =
-        new SegmentProcessorConfig.Builder().setTableConfig(tableConfig).setSchema(schema)
-            .setMergeType(MergeType.CONCAT);
-    if (!aggregationTypes.isEmpty()) {
-      segmentProcessorConfigBuilder.setAggregationTypes(aggregationTypes);
-    }
-    if (numRecordsPerSegmentString != null) {
-      segmentProcessorConfigBuilder.setSegmentConfig(
-          new SegmentConfig.Builder().setMaxNumRecordsPerSegment(Integer.parseInt(numRecordsPerSegmentString)).build());
-    }
+        new SegmentProcessorConfig.Builder().setTableConfig(tableConfig).setSchema(schema);
+
+    // Time handler config
+    segmentProcessorConfigBuilder
+        .setTimeHandlerConfig(MergeTaskUtils.getTimeHandlerConfig(tableConfig, schema, configs));
+
+    // Partitioner config
```

Comment:
For handling the custom partitioning, are we going to depend on the segment framework or the task scheduler?
One way to handle this is to make merge segments from the same partition. Another way is to do the partitioning based on the underlying partitioning column value in the framework.

Reply:
Currently we don't have the support for custom partitioning, and only partition on time and the partition column within the table config.

Reply:
I meant the partition column within the table config as custom partitioning.

Reply:
I see. The segment framework will always partition the data in case the old segment is not properly partitioned. The overhead should be fairly low.